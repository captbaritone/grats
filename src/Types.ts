import type { GraphQLResolveInfo } from "graphql";

/** @gqlScalar */
export type Float = number;
/** @gqlScalar */
export type Int = number;
/** @gqlScalar */
export type ID = string;

/** @gqlInfo */
export type GqlInfo = GraphQLResolveInfo;
