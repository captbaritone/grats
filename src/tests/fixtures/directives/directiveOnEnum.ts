import { Int } from "../../../Types";
/**
 * This is my custom directive.
 * @gqlDirective on ENUM
 */
export function max(args: { foo: Int }) {}

/**
 * @gqlEnum
 * @gqlAnnotate max(foo: 10)
 */
type MyEnum = "A" | "B";
