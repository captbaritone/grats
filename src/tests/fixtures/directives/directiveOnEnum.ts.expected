-----------------
INPUT
----------------- 
import { Int } from "../../../Types";
/**
 * This is my custom directive.
 * @gqlDirective on ENUM
 */
export function max(args: { foo: Int }) {}

/**
 * @gqlEnum
 * @gqlAnnotate max(foo: 10)
 */
type MyEnum = "A" | "B";

-----------------
OUTPUT
-----------------
-- SDL --
"""This is my custom directive."""
directive @max(foo: Int!) on ENUM

enum MyEnum @max(foo: 10) {
  A
  B
}
-- TypeScript --
import { GraphQLSchema, GraphQLDirective, DirectiveLocation, GraphQLNonNull, GraphQLInt, GraphQLEnumType } from "graphql";
export function getSchema(): GraphQLSchema {
    const MyEnumType: GraphQLEnumType = new GraphQLEnumType({
        name: "MyEnum",
        values: {
            A: {
                value: "A"
            },
            B: {
                value: "B"
            }
        },
        extensions: {
            grats: {
                directives: [{
                        name: "max",
                        args: {
                            foo: 10
                        }
                    }]
            }
        }
    });
    return new GraphQLSchema({
        directives: [new GraphQLDirective({
                name: "max",
                locations: [DirectiveLocation.ENUM],
                description: "This is my custom directive.",
                args: {
                    foo: {
                        type: new GraphQLNonNull(GraphQLInt)
                    }
                }
            })],
        types: [MyEnumType]
    });
}
